pyproject.toml

```diff
@@ -102,7 +102,14 @@ test = [
 ]
 gpu = [
     "cupy",
-    "numpy<2.0.0",
+]
+
+cu12 = [
+    "cupy-cuda12x",
+]
+
+cu11 = [
+    "cupy-cuda11x",
 ]
```

Comment:
maybe we shouldn’t have an extra here at all and just document how to get GPU support?

Reply:
we could also throw out the complete cupy setup and just add docs for it

Reply:
Well, I was opposed to having both that _and_ a `gpu` extra.  But I would be fine with @flying-sheep's solution if we get rid of `gpu`.  My only thing would be if we add another library in the future for GPU.  @ivirshup might have insight as to what could come.  Maybe `cudf`?